### DISCUSSION TITLE
My suggestions for improvemen

### DISCUSSION THREAD

**Discussion by webdsf** [mediacms-io/mediacms#35] [Feature Requests]
1. How to force download not only mkv and mp4? I would like to add avi, vob.
2. Language packs, let users translate, you just give the opportunity. I can translate into Russian.
3. The problem is with Russian, there is no way to upload files that differ from English characters. And there is no way to change the name of the video other than English.  Also, I cannot change the description. _Server Error (500)_
4. Possible SubRip support, or transcoding at boot. (also the ability to get subtitles from the videocontainer)
5. Additional customization of encoding profiles, and adding new ones. Ability to bring out the settings for quality compression, sound directly from the admin panel. For easy creation of profiles for unknowing people.
6. For some reason, the vp9 profile does not work, I selected it, but the video is not encoded into it.
Someday it will be good to add support for ads, additional audio tracks, choice of storage space, storage in the clouds and ftp, prevent the inserted video from playing on certain domains.

I really liked the script, it works well for me with a weak VPS. And the quality is decent. There are simply not enough opportunities, I would even like to help as much as I can. I see potential here, and I believe in you! With the help of your script will be doing a lot of great-million projects!

**Comment by mgogoulos** [comment]
Hello, 
1) you mean uploading video files like avi? They are supported
2) planned for the near future
3) that's a bug, can confirm it, will work on a solution
5) possible, through the admin dashboard (/admin/files/encodeprofile/) , you can see what is available and enable/disable. To add new profiles however would require more work than adding them there. 
6) lots of good ideas here!

**Reply by webdsf** [reply]
1.I figured it out, since I turned off the profile 240,360, 480 it was not encoded. I just wanted the maximum quality from the same avi and VOB. Not like YouTube that spoiled without realizing that it was possible to draw out the best quality.
2. I'm really looking forward, otherwise wanted to do it myself)
3. [Video](https://youtu.be/dg-kiy5Rf5c), [Log](https://drive.google.com/drive/folders/1ssfBd-F2cpjxdn2UMg0mZKaIqXI7IT9b?usp=sharing) I can still throw off other information. Also carry out experiments.
4. No, don't add new. Change characteristics. How much to compress, what audio and sound quality.
Also interested. how can you adjust the split flow?

**Reply by webdsf** [reply]
Another would be with subtitles the ability to upload srt, or recode on the fly when uploading srt to the site. It is very inconvenient to encode them separately using a service.

**Reply by mgogoulos** [reply]
Can you try uploading that video to demo.mediacms.io, and ping me if it breaks there too? Cannot figure out what is the issue looking on the logs. i thought I could reproduce the issue but ends up I was wrong. It looks like an encoding issue, let's see

**Reply by webdsf** [reply]
Everything works in your version, great! So when renaming, and when loading a video with Russian characters.
The same approach will apply well to other languages as well. And internationality will be ensured. And when you add languages.
I also think it will be decided somehow later, and at the end of the name, the video extension adds.
 Wow, waiting for your new lines of code!
